ActionPlans/Start-DlToO365GroupUpgradeChecks.ps1

```diff
@@ -303,45 +303,61 @@ else {
 }
 #endregion Check if Distribution Group can't be upgraded because Distribution list which has more than 100 owners or it has no owner
 
-#new condition to check if owner has mailbox or not
-#region Check if Distribution Group can't be upgraded because anyone of the owners doesn't have a mailbox
+#region Check if Distribution Group can't be upgraded because the distribution list owner(s) is non-supported with RecipientTypeDetails other than UserMailbox, MailUser
 if ($checkifownerhasmailbox -match "Continuechecking")
 {
-    [string]$SectionTitle = "Validating Distribution Group Owners Mailbox Status"
-    [string]$Description = "Checking if Distribution Group can't be upgraded because one or more DL owners doesn't have a mailbox in Exchange Online"
-    $ConditionDGownerswithoutMBX=@()
+    [string]$SectionTitle = "Validating Distribution Group Owners RecipientTypeDetails"
+    [string]$Description = "Checking if Distribution Group can't be upgraded because DL owner(s) is non-supported with RecipientTypeDetails other than UserMailbox, MailUser"
+    $ConditionDGownernonsupported=@()
+    $ConditionDGownernonsupportedforusers=@()
     foreach($owner in $owners)
     {
         try {
-            $owner=Get-User $owner -ErrorAction stop
-            if ($owner.RecipientType -eq "User") 
+            $owner=Get-Recipient $owner -ErrorAction stop
+            if ($owner.RecipientTypeDetails -ne "UserMailbox" -and $owner.RecipientTypeDetails -ne "MailUser") 
                 { 
-                    $ConditionDGownerswithoutMBX=$ConditionDGownerswithoutMBX+$owner
+                    $ConditionDGownernonsupported=$ConditionDGownernonsupported+$owner
                 }
         }
         catch {
-            $CurrentProperty = "Validating: $owner mailbox status"
+            $CurrentProperty = "Validating: $owner RecipientTypeDetails"
             $CurrentDescription = "Failure"
-            write-log -Function "Validate owner mailbox status" -Step $CurrentProperty -Description $CurrentDescription
+            write-log -Function "Validate owner RecipientTypeDetails" -Step $CurrentProperty -Description $CurrentDescription
+            #check if the owner RecipientTypeDetails is User
+            $owner=Get-User $owner -ErrorAction stop
+            $ConditionDGownernonsupportedforusers=$ConditionDGownernonsupportedforusers+$owner
         }
     }
-    if($ConditionDGownerswithoutMBX.Count -ge 1)
+    if($ConditionDGownernonsupported.Count -ge 1 -and $ConditionDGownernonsupportedforusers.Count -ge 1)
```

Comment:
Here you are checking to have entries in both (non MailUser and UserMailbox) AND (not a user or recipient).
It will be true only if you have entries in both list.
Please re-evaluate the condition

Reply:
Am having 2 lists one for the output of get-recipient "ConditionDGownernonsupported" & 2nd one of the output of get-user "ConditionDGownernonsupportedforusers"
Am checking if the 2 lists are including users inside in case of DL Owner RecipientTypeDetails "e.g. MailContact & User" or any of the 2 lists are including users inside  "e.g. MailContact" OR "e.g. User" else then both lists are empty!

 if($ConditionDGownernonsupported.Count -ge 1 -and $ConditionDGownernonsupportedforusers.Count -ge 1)
 elseif ($ConditionDGownernonsupported.Count -ge 1 -and $ConditionDGownernonsupportedforusers.Count -lt 1)
elseif ($ConditionDGownernonsupported.Count -lt 1 -and $ConditionDGownernonsupportedforusers.Count -ge 1) 
else